dashboard/app/models/course.rb

```diff
@@ -362,6 +362,30 @@ def select_course_script(user, default_course_script)
     default_course_script
   end
 
+  # If the user does not have progress on any script in the current course, but
+  # has progress on a script in another course in the same course family, then
+  # they are on the wrong version of the course.
+  # @param user [User]
+  # @return [Boolean] Whether the user is on the wrong version of the course.
+  def wrong_version?(user)
+    return nil unless user
+    user_script_ids = user.user_scripts.pluck(:script_id)
```

Comment:
We also create user script for each student when a script is assigned to a section, so if they were ever assigned to a specific script, that would look like progress and the script would be included here. That might still be ok, just something to consider.

Reply:
yes, although I call it "progress", we want to show this warning if a student has been assigned to (rather than made progress in) a different version of the course. I'd be happy to use a different word for "started or made progress in or was assigned to" if we have one.